modules/core/04-channel/keeper/packet_test.go

```diff
@@ -322,35 +322,35 @@ func (suite *KeeperTestSuite) TestRecvPacket() {
 			// attempts to receive packet 2 without receiving packet 1
 			channelCap = suite.chainB.GetChannelCapability(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
 		}, true},
-		// {
-		// 	"success with channel in ACKUPGRADE: FLUSHING status",
-		// 	func() {
-		// 		suite.coordinator.Setup(path)
-		// 		sequence, err := path.EndpointA.SendPacket(defaultTimeoutHeight, disabledTimeoutTimestamp, ibctesting.MockPacketData)
-		// 		suite.Require().NoError(err)
-		// 		packet = types.NewPacket(ibctesting.MockPacketData, sequence, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, defaultTimeoutHeight, disabledTimeoutTimestamp)
-		// 		channelCap = suite.chainB.GetChannelCapability(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
-
-		// 		// Move channel to correct state.
-		// 		path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = ibcmock.UpgradeVersion
-
-		// 		err = path.EndpointB.ChanUpgradeInit()
-		// 		suite.Require().NoError(err)
-
-		// 		err = path.EndpointA.ChanUpgradeTry()
-		// 		suite.Require().NoError(err)
-
-		// 		err = path.EndpointB.ChanUpgradeAck()
-		// 		suite.Require().NoError(err)
-
-		// 		channel := path.EndpointB.GetChannel()
-		// 		channel.FlushStatus = types.FLUSHING
-		// 		path.EndpointB.SetChannel(channel)
-
-		// 		suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.SetCounterpartyLastPacketSequence(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, sequence+1)
-		// 	},
-		// 	true,
-		// },
+		{
+			"success with channel in Flushing status",
+			func() {
+				suite.coordinator.Setup(path)
+				sequence, err := path.EndpointA.SendPacket(defaultTimeoutHeight, disabledTimeoutTimestamp, ibctesting.MockPacketData)
+				suite.Require().NoError(err)
+				packet = types.NewPacket(ibctesting.MockPacketData, sequence, path.EndpointA.ChannelConfig.PortID, path.EndpointA.ChannelID, path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, defaultTimeoutHeight, disabledTimeoutTimestamp)
+				channelCap = suite.chainB.GetChannelCapability(path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID)
+
+				// Move channel to correct state.
+				path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = ibcmock.UpgradeVersion
+
+				err = path.EndpointB.ChanUpgradeInit()
+				suite.Require().NoError(err)
+
+				err = path.EndpointA.ChanUpgradeTry()
+				suite.Require().NoError(err)
+
+				err = path.EndpointB.ChanUpgradeAck()
+				suite.Require().NoError(err)
+
+				channel := path.EndpointB.GetChannel()
+				channel.State = types.STATE_FLUSHING
+				path.EndpointB.SetChannel(channel)
+
+				suite.chainB.GetSimApp().IBCKeeper.ChannelKeeper.SetCounterpartyLastPacketSequence(suite.chainB.GetContext(), path.EndpointB.ChannelConfig.PortID, path.EndpointB.ChannelID, sequence+1)
```

Comment:
aha yea! based on the refactor of the code I dumped in the other comment this could be removed, right?

Right now this manual setting is required to get tests to pass?

Reply:
Just some food for thought - It might be more maintainable to directly set the chainB endpoint channel's state to `FLUSHING` and set the counterparty upgrade in its store with an appropriate packet sequence, rather than relying on doing the full upgrade flow with the endpoint helpers.

Reply:
Went with setting it directly for now. I do recall a comment on another PR mentioning it would be nicer to use handshake to go to right state but we can decide on that after all changes have been finalized.

Reply:
Yeah, I mean, there's pros and cons to both. I personally prefer setting directly as its easier to maintain the tests in isolation. A lot of the existing tests used the endpoint helpers doing the full handshake and when we came to start this refactor I found that it ended up breaking so many tests because they relied on the full handshake